### PR TITLE
Admin improvs

### PR DESCRIPTION
**Pull request by chrisrodz**
### More improvements to the admin

@MrMaksimize I have some nice work for you to review courtesy of @clarita1980 
#### Summary of things changed
- [x] Added Working Capital option to Purpose choices
- [x] Changed purpose to Array(Text) field
- [x] Changed age to Array(Int) field
- [x] Changed benefitType to Array(Text) field
- [x] Improved front-end field emptiness validation
#### Files changed
- lib/options.js
  - Add `working_capital` to `purpose` dictionary
- models/defs/Opportunity.js
  - Changed `DataType` to `Array` for `purpose`, `age` and `benefitType`
  - Changed `widget` to `multiSelect` for `purpose`, `age` and `benefitType`
  - Added field `multiple: true` for `purpose`, `age` and `benefitType`
- models/lib/modelUtils.js
  - Updated other text fields name from `-other` to `Other`
- public/src/js/main.js
  - Removed checkbox emptiness validation for admin models
  - Added select2 emptiness validations for admin models
- test/mocks/opportunity.js
  - Updated mock other field names from `-other` to `Other`
  - Updates mock fields `purpose`, `age` and `benefitType` to be arrays.
- test/mocks/queryForm.js
  - Updated mock other field names from `-other` to `Other`
- test/models.js
  - Updated tests to consider migrated array fields
- views/mixins/forms.jade
  - Removed `required` attribute from other fields
  - Changed other field names from `-other` to `Other`
#### Files Added
- migrations/...add-more-multi-select-fields.js
  - Migration from callback hell!

### CODE REVIEW

File: migrations/20140711105052-add-more-multi-select-fields.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  up: function(migration, DataTypes, done) {    
+    migration.addColumn(
+      'opportunities',
+      'purposeTemp',
+      {
+        type: DataTypes.ARRAY(DataTypes.TEXT),
+        allowNull: false,
+        widget: 'multiSelect',
+        choices: choicesList.getFormChoices('purpose'),
+        validate: {
```
Comment (MrMaksimize) [inline migrations/20140711105052-add-more-multi-select-fields.js:13]: Maybe dump the empty object? Or condense to one line?  I think the default is {} anyway

File: migrations/20140711105052-add-more-multi-select-fields.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        allowNull: false,
+        widget: 'multiSelect',
+        choices: choicesList.getFormChoices('benefitType'),
+        label: 'Benefit Type',
+        defaultValue: ['incentive'],
+        choiceOther: true
+      }
+    );
+    var Model = migration.migrator.sequelize.model('opportunity');
+    Model.all().success(function(result) {
```
Comment (MrMaksimize) [inline migrations/20140711105052-add-more-multi-select-fields.js:46]: This is like...serious

**Comment by MrMaksimize** [comment]
Whoa.... This is serious... Looks good but i wanna do another quick review tomorrow before merging :)

**Comment by MrMaksimize** [comment]
Looked through again.  Awesome work @chrisrodz!! Ran a backup for safety on production, but merging this into staging :)

**Comment by chrisrodz** [comment]
Awesome!!

On Monday, July 14, 2014, Maksim Pecherskiy notifications@github.com
wrote:

> Merged #27 https://github.com/CoquiCoders/bizwallet/pull/27.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CoquiCoders/bizwallet/pull/27#event-141301572.

## 

Best,
Christian A. Rodriguez

**Pull request merged**